### PR TITLE
Update to be compatible with SS4

### PR DESCRIPTION
**Pull request by andrewandante**
Namespaces and structure altered appropriately to be ready for SS4.

### CODE REVIEW

**Comment by jonom** [comment]
Thanks for this! Could I bug you to do two extra things please:
1) Make the namespace JonoM instead of Jonom
2) Update the editor config for 4 space indentation for composer.json

**Comment by andrewandante** [comment]
No problem. Have switched the namespace to `JonoM`. Think that `composer.json` should get caught by the wildcard which is set to 4 spaces [here](https://github.com/jonom/silverstripe-betternavigator/blob/master/.editorconfig#L8-L9). In fact, that's probably why PHPStorm auto-corrected it!

**Comment by jonom** [comment]
Thanks @andrewandante I'm on vacation but I'll try to find a sec to test this soon and get it merged

**Comment by jonom** [comment]
Sorry 😬  can you please re-base? Then I promise I'll test it out.

**Comment by andrewandante** [comment]
Gone done a rebase - we might need to remove the config line now, but we'll see. I've been using it so I'm confident :)

File: src/Extension/BetterNavigatorExtension.php [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\Security\Permission;
+use SilverStripe\CMS\Controllers\SilverStripeNavigator;
+use SilverStripe\Control\Director;
+use SilverStripe\Security\Member;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Control\Controller;
+use SilverStripe\View\ArrayData;
+use SilverStripe\ORM\Versioning\Versioned;
```
Comment (jonom) [inline src/Extension/BetterNavigatorExtension.php]: This has been split off so needs to be changed to `SilverStripe\Versioned\Versioned`

**Comment by andrewandante** [comment]
Alright @jonom , today is hackfest day and I've finally got around to doing it. Have fixed both the versioning problem and the editlink one. Let me know!

File: src/Extension/BetterNavigatorExtension.php [hunk 2]
```diff
@@ -9,19 +22,20 @@ class BetterNavigatorExtension extends DataExtension {
      * @return string
      */
     public function BetterNavigator() {
+
         // Make sure this is a page
         if (!($this->owner && $this->owner->dataRecord && $this->owner->dataRecord instanceof SiteTree && $this->owner->dataRecord->ID > 0)) return false;
```
Comment (Firesphere) [inline src/Extension/BetterNavigatorExtension.php]: Not part of this commit, but whoa that's a big if!

File: src/Extension/BetterNavigatorExtension.php [hunk 3]
```diff
@@ -9,19 +22,20 @@ class BetterNavigatorExtension extends DataExtension {
...
 
         // Only show navigator to appropriate users
         $isDev = Director::isDev();
         $canViewDraft = (Permission::check('VIEW_DRAFT_CONTENT') || Permission::check('CMS_ACCESS_CMSMain'));
-        if($isDev || $canViewDraft) {
+        if ($isDev || $canViewDraft) {
             // Get SilverStripeNavigator links & stage info (CMS/Stage/Live/Archive)
             $nav = array();
             $viewing = '';
             $navigator = new SilverStripeNavigator($this->owner->dataRecord);
```
Comment (Firesphere) [inline src/Extension/BetterNavigatorExtension.php]: It's not an Object::create() option?

Comment (andrewandante) [inline src/Extension/BetterNavigatorExtension.php]: Will do a little code tidying then :)

**Comment by jonom** [comment]
Thanks @andrewandante!
